src/main/java/com/iota/iri/service/API.java

```diff
@@ -1730,4 +1801,53 @@ private synchronized AbstractResponse storeMessageStatement(String address, Stri
         return elements;
     }
 
+    // ------ permanode ----------
+
+    private Function<Map<String, Object>, AbstractResponse> pinTransactionTrytes() {
+        return request -> {
+            final List<String> transactionTrytes = getParameterAsList(request,"trytes", TRYTES_SIZE);
```

Comment:
I think adding `final` in a method scope is rarely helpful.
You still have a mutable list anyhows.

`getParameterAsList` should return an immutable list. But this is not in the scope of your PR :-)

```suggestion
            List<String> transactionTrytes = getParameterAsList(request,"trytes", TRYTES_SIZE);
```

Reply:
I had the same feeling but I just copied the rest of the work in the API.java, which uses final lists for almost all methods. Just following the code-style here ;)